x/archivedb/db.go

```diff
@@ -121,64 +118,6 @@ func (db *archiveDB) Get(key []byte, height uint64) ([]byte, uint64, error) {
 }
 
 // Creates a new batch to append database changes in a given height
-func (db *archiveDB) NewBatch() (batchWithHeight, error) {
-	var nextHeightBytes [8]byte
-
-	batch := db.rawDB.NewBatch()
-	nextHeight := db.currentHeight + 1
-	binary.BigEndian.PutUint64(nextHeightBytes[:], nextHeight)
-	err := batch.Put([]byte(currentHeightKey), nextHeightBytes[:])
-	if err != nil {
-		return batchWithHeight{}, err
-	}
-	return batchWithHeight{
-		db:     db,
-		height: nextHeight,
-		batch:  batch,
-	}, nil
-}
-
-func (c *batchWithHeight) Height() uint64 {
-	return c.height
-}
-
-// Writes the changes to the database
-func (c *batchWithHeight) Write() error {
-	err := c.batch.Write()
-	if err != nil {
-		return err
-	}
-	c.db.currentHeight = c.height
-	return nil
-}
-
-// Delete any previous state that may be stored in the database
-func (c *batchWithHeight) Delete(key []byte) error {
-	internalKey := newInternalKey(key, c.height)
-	internalKey.isDeleted = true
-	return c.batch.Put(internalKey.Bytes(), []byte{})
-}
-
-// Queues an insert for a key with a given
-func (c *batchWithHeight) Put(key []byte, value []byte) error {
-	internalKey := newInternalKey(key, c.height)
-	return c.batch.Put(internalKey.Bytes(), value)
-}
-
-// Returns the sizes to be committed in the database
-func (c *batchWithHeight) Size() int {
-	return c.batch.Size()
-}
-
-// Removed all pending writes and deletes to the database
-func (c *batchWithHeight) Reset() {
-	c.batch.Reset()
-}
-
-func (c *batchWithHeight) Inner() database.Batch {
-	return c.batch
-}
-
-func (c *batchWithHeight) Replay(w database.KeyValueWriterDeleter) error {
-	return c.batch.Replay(w)
+func (db *archiveDB) NewBatch(height uint64) *dbBatchWithHeight {
```

Comment:
Are we trying to satisfy the `database.Database` interface eventually? Because this method signature isn't compatible with that interface

Reply:
One way to implement the interface if we wanted would to eg have a method like `NewBatcher(height uint64)`.
We can change this later as well.